packages/math/src/pool/concentrated/utils.ts

```diff
@@ -0,0 +1,137 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+
+import { smallestDec } from "./const";
+
+export function calcAmount0Delta(
+  liquidity: Dec,
+  sqrtPriceA: Dec,
+  sqrtPriceB: Dec,
+  roundUp = true
+): Dec {
+  if (sqrtPriceA.gt(sqrtPriceB)) {
+    [sqrtPriceA, sqrtPriceB] = [sqrtPriceB, sqrtPriceA];
+  }
+  const diff = sqrtPriceB.sub(sqrtPriceA);
+  const denom = sqrtPriceA.mul(sqrtPriceB);
+
+  if (roundUp) {
+    return liquidity.mul(diff).quo(denom).roundUpDec();
+  }
+  return liquidity.mul(diff).quo(denom);
+}
+
+export function calcAmount1Delta(
+  liquidity: Dec,
+  sqrtPriceA: Dec,
+  sqrtPriceB: Dec,
+  roundUp = true
+): Dec {
+  if (sqrtPriceA.gt(sqrtPriceB)) {
+    [sqrtPriceA, sqrtPriceB] = [sqrtPriceB, sqrtPriceA];
+  }
+  const diff = sqrtPriceB.sub(sqrtPriceA);
+
+  if (roundUp) {
+    return liquidity.mul(diff).roundUpDec();
+  }
+  return liquidity.mul(diff);
+}
+
+export function getNextSqrtPriceFromAmount0RoundUp(
```

Comment:
Note: to support swapping in given out, there will be more helpers for computing next sqrt price. PRin progress: https://github.com/osmosis-labs/osmosis/pull/4477